test/functional/plugin/lsp_spec.lua

```diff
@@ -493,6 +493,31 @@ describe('LSP', function()
       eq(0, result.num_attached_after)
     end)
 
+    it('should not re-attach buffer if it was deleted in on_init', function()
```

Comment:
```suggestion
    it('should not re-attach buffer if it was deleted in on_init #28575', function()
```

However, I find the test does not fail without the patch on `vim.lsp.client`.